src/pretix/control/views/waitinglist.py

```diff
@@ -160,10 +160,15 @@ def post(self, request, *args, **kwargs):
                 'forbidden': self.get_queryset().filter(voucher__isnull=False),
             })
         elif request.POST.get('action') == 'delete_confirm':
-            for obj in self.get_queryset(force_filtered=True):
-                if not obj.voucher_id:
-                    obj.log_action('pretix.event.orders.waitinglist.deleted', user=self.request.user)
-                    obj.delete()
+            with transaction.atomic():
+                log_entries = []
+                to_delete = []
+                for obj in self.get_queryset(force_filtered=True):
+                    if not obj.voucher_id:
+                        log_entries.append(obj.log_action('pretix.event.orders.waitinglist.deleted', user=self.request.user, save=False))
+                        to_delete.append(obj.pk)
+                LogEntry.bulk_create_and_postprocess(log_entries)
+                WaitingListEntry.objects.filter(id__in=to_delete).delete()
```

Comment:
Probably does not make difference, but in the above code-parts we first do the change and then write the log. Maybe a micro-optimization regarding the transaction, but I guess if something fails it is in the delete-call and that way, the log-entries would not have to rollback?